app/src/org/commcare/CommCareApplication.java

```diff
@@ -275,9 +279,42 @@ protected void turnOnStrictMode() {
     }
 
     @Override
-    public void onConfigurationChanged(Configuration newConfig) {
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
         LocalePreferences.saveDeviceLocale(newConfig.locale);
+        updateLocale(this);
+    }
+
+    private void updateLocale(Context context) {

```

Comment:
Is this method needed? Seems like the OS properly updates views when the user switches languages, even if I comment this out.

Reply:
Yes, you're right. The OS automatically updates the view after changing the language from the system settings without needing to set the locale in the code.

Reply:
Code updated